File: db/transport.php
<?php

require_once("entity.php");

/**
 * Class for accessing, manipulating, creating and deleting records in transport.
*/

class Transport extends Entity
{

  public function __construct($select = array())
  {
    $this->table = "transport";
    $this->order = "";
    $this->domain = "valuelist";
    $this->field['transport_id']['type'] = 'SERIAL';
    $this->field['transport_id']['not_null'] = true;
    $this->field['tag']['type'] = 'VARCHAR';
    $this->field['tag']['length'] = 32;
    $this->field['tag']['not_null'] = true;
    $this->field['rank']['type'] = 'INTEGER';
    $this->field['transport_id']['primary_key'] = true;
    parent::__construct($select);
  }

}

?>

File: pentabarf/lib/momomoto/tables/transport.rb
module Momomoto
  class Transport < Base
    def initialize
      super
      @domain = 'valuelist'
      @fields = {
        :transport_id => Datatype::Integer.new( {:not_null=>true, :default=>true, :primary_key=>true, :serial=>true} ),
        :tag => Datatype::Varchar.new( {:not_null=>true, :length=>32} ),
        :rank => Datatype::Integer.new( {} )
      }
    end
  end
end
